### PR TITLE
feat(api): add perform module fw update endpoint

### LINKED ISSUE #4576: App: show and proceed with available module updates
## overview

This provides the front-end changes to #4575.
The currently designed "Updated" button in the Modules & Pipettes tab should suffice for an initial UI. 

In the GET /modules response, parse out whether or not there is an available update for a given module by serial number. If there is, enable the button and change its text to "Update".

Upon clicking, show some loading state (preferrably one that states "keep module plugged in and turned on"), and display the module update response.

### PR DESCRIPTION
**Pull request by b-cooper**
## overview

Add ability to initiate a module fw update for a given attached module, referenced by its serial
number via http. Remove old unused module os level configuration. Update the module update machinery
and comments to be more accurate and generic.

Closes #4576

## changelog

  - add support for POST `/modules/[SERIAL]/update`, with no body
  - cleanup access to avrdude configuration, should be all through buildroot now
  - reorganize some of the module update logic to be more generic and accurate

## review requests

- Make sure your robot has a new buildroot build that includes module firmware files in `/usr/lib/firmware/` 
- Attached modules of each type [tc, md, td] to the robot that have lower firmware versions than those listed in the file names of `/usr/lib/firmware`
- POST to `[ROBOT_IP]:31950/modules/[MODULE_SERIAL]/update` and confirm that the fw update succeeds

### CODE REVIEW

File: api/src/opentrons/server/endpoints/update.py [hunk 1]
```diff
@@ -1,71 +1,74 @@
...
+    """
+    log.debug('Update Module Firmware request received')
+    serial = request.match_info['serial']
 
-async def _upload_to_module(hw, serialnum, fw_filename, loop):
-    hw_mods = hw.attached_modules
-    for module in hw_mods:
-        if module.device_info.get('serial') == serialnum:
-            log.info("Module with serial {} found".format(serialnum))
+    for module in request.app['com.opentrons.hardware'].attached_modules:
```
Comment (sfoster1) [inline api/src/opentrons/server/endpoints/update.py]: let's make this return early if it doesn't find the module, we can cut a level of indentation that way

File: api/src/opentrons/server/endpoints/update.py [hunk 2]
```diff
@@ -1,71 +1,74 @@
...
+                    await asyncio.wait_for(
+                        modules.update_firmware(
+                            module, module.bundled_fw.path, request.loop),
+                        UPDATE_TIMEOUT)
+                    res = f'Successully updated module {serial}'
+                    status = 200
+                else:
+                    res = (f'Bundled fw file not found for module of '
+                           f'type: {module.name()}')
+                    status = 404
```
Comment (sfoster1) [inline api/src/opentrons/server/endpoints/update.py]: this doesn't quite seem like the appropriate status code? but i'm not sure what would be better

File: api/src/opentrons/server/endpoints/update.py [hunk 3]
```diff
@@ -1,71 +1,74 @@
...
             except modules.UpdateError as e:
-                return f'Bootloader error: {e}', 400
+                res = f'Bootloader error: {e}'
+                status = 400
```
Comment (sfoster1) [inline api/src/opentrons/server/endpoints/update.py]: this probably should be a 500 also, considering the request doesn't provide any info other than which module to update it's hard to see how it should be a 4xx

**Review by sfoster1 — CHANGES_REQUESTED** [review]
Couple nitpicks but it looks good overall

File: api/src/opentrons/server/endpoints/update.py (hunk 2 above)
Comment (b-cooper) [reply api/src/opentrons/server/endpoints/update.py]: I'll change it to 500

**Review by sfoster1 — APPROVED** [review]
LGTM

**Review by mcous — DISMISSED** [review]
Can you also add this endpoint to HAPIv1's OpenAPI spec? `api/src/opentrons/server/openapi.json`

**Review by sfoster1 — APPROVED** [review]
Rereviewed the module object refresh. Looks good.

**Pull request merged**